From a7b5cfc144065a68d53d9818cab1183955fb5fbb Mon Sep 17 00:00:00 2001
From: Jerome Forissier <jerome.forissier@linaro.org>
Date: Thu, 5 Jul 2018 17:13:15 +0200
Subject: [PATCH 06/10] tee-supplicant: rpmb.c: add
 __attribute__((fallthrough))
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix the following error reported by GCC 8.1:

src/rpmb.c: In function ‘rpmb_data_req’:
src/rpmb.c:683:6: error: this statement may fall through [-Werror=implicit-fallthrough=]
   if (rsp_nfrm != 1) {
      ^
src/rpmb.c:689:2: note: here
  case RPMB_MSG_TYPE_REQ_AUTH_DATA_READ:
  ^~~~

I could not silence this warning using any of the special comments
documented on the GCC Warning Options page [1]. It is weird, because we
use -Wextra which is supposed to enable -Wimplicit-fallthrough=3, which is
supposed to recognize the comment we had already (/* Fall through */).

Link: [1] https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html
Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>
---
 tee-supplicant/src/rpmb.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/tee-supplicant/src/rpmb.c b/tee-supplicant/src/rpmb.c
index 3246cd9..fe45b69 100644
--- a/tee-supplicant/src/rpmb.c
+++ b/tee-supplicant/src/rpmb.c
@@ -713,7 +713,6 @@ static uint32_t rpmb_data_req(int fd, struct rpmb_data_frame *req_frm,
 		__attribute__((fallthrough));
 #endif
 
-		/* Fall through */
 	case RPMB_MSG_TYPE_REQ_AUTH_DATA_READ:
 		mioc->num_of_cmds = 2;
 
-- 
2.17.1

